libbpf-tools/sysinjector.bpf.c

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright 2021 Sony Group Corporation */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "sysinjector.h"
+
+const volatile pid_t targ_tgid = 0;
+const volatile char targ_comm[TASK_COMM_LEN] = {};
+const volatile int retval = 0;
+
+static __always_inline int is_same_comm(const char *comm)
+{
+	int i;
+	for (i = 0; targ_comm[i] != '\0' && i < TASK_COMM_LEN; i++) {
+		if (comm[i] != targ_comm[i])
+			return false;
+	}
+	return true;
```

Comment:
targ_comm = `abc` and comm = `abcd` will be evaluated as true.

Reply:
I'll reuse comm_allowed().

Reply:
The case above not yet resolved.